app/api/vercel/webhook/route.ts

```diff
@@ -0,0 +1,219 @@
+  meta: Record<string, string>;
+  url: string;
+  name: string;
+}
+
+interface VercelDeploymentEvent {
+  type: string;
+  payload: string | { text?: string };
+}
+
+interface VercelWebhookPayload {
+  type: string;
+  id: string;
+  createdAt: string;
+  region: string;
+  payload: {
+    team?: { id: string };
+    user: { id: string };
+    deployment: VercelDeployment;
+    links: {
+      deployment: string;
+      project: string;
+    };
+    target: string;
+    project: { id: string };
+    plan: string;
+    regions: string[];
+  };
+}
+
+export async function POST(request: Request) {
+  try {
+    const body = await request.text();
+    const vercelWebhookSecret = process.env.VERCEL_WEBHOOK_SECRET;
+    
+    if (!vercelWebhookSecret) {
+      console.error("VERCEL_WEBHOOK_SECRET environment variable is not set");
+      return NextResponse.json(
+        { error: "Vercel webhook configuration missing" },
+        { status: 500 }
+      );
+    }
+
+    const vercelSignature = request.headers.get("x-vercel-signature");
+    
+    if (!vercelSignature) {
+      console.error("Missing x-vercel-signature header");
+      return NextResponse.json(
+        { error: "Missing x-vercel-signature header" },
+        { status: 400 }
+      );
+    }
+
+    const hmac = createHmac("sha1", vercelWebhookSecret);
+    hmac.update(body);
+    const expectedSignature = hmac.digest("hex");
+    
+    if (vercelSignature !== expectedSignature) {
+      console.error("Vercel webhook signature verification failed");
+      return NextResponse.json(
+        { error: "Webhook signature verification failed" },
+        { status: 400 }
+      );
+    }
+
+    const payload = JSON.parse(body) as VercelWebhookPayload;
+    const eventType = payload.type;
+
+    console.log(`Received Vercel webhook event: ${eventType}`);
+
+    if (eventType === "deployment.error") {
+      await handleDeploymentError(payload);
+    } else {
+      console.log(`Unhandled event type: ${eventType}`);
+    }
+
+    return NextResponse.json({ received: true });
+  } catch (error) {
+    console.error("Error processing Vercel webhook:", error);
+    return NextResponse.json(
+      { error: "Failed to process Vercel webhook" },
+      { status: 500 }
+    );
+  }
+}
+
+async function handleDeploymentError(payload: VercelWebhookPayload) {
+  console.log("Deployment error:", {
+    deploymentId: payload.payload.deployment.id,
+    deploymentUrl: payload.payload.deployment.url,
+    dashboardLink: payload.payload.links.deployment,
+    target: payload.payload.target
+  });
+
+  try {
+    const prNumber = extractPRNumber(payload.payload.deployment.meta);
+    
+    if (!prNumber) {
+      console.log("No PR number found in deployment metadata, skipping PR comment");
+      return;
+    }
+
+    const buildLogs = await fetchVercelBuildLogs(payload.payload.deployment.id);
+    
+    const comment = formatBuildLogsComment(payload, buildLogs);
+    
+    await postGitHubComment(
```

Comment:
PostGithubComment relies on GITHUB_TOKEN being available, which we won't have. Get a GitHub token using GITHUB_PRIVATE_KEY